src/ape/plugins/__init__.py

```diff
@@ -145,26 +153,33 @@ def get_plugin_name_and_hookfn(h):
             return h.plugin_name, getattr(h.plugin, attr_name)()
 
         for plugin_name, results in map(get_plugin_name_and_hookfn, hookimpls):
+
             # NOTE: Some plugins return a tuple and some return iterators
-            if not isinstance(results, tuple) and hasattr(results, "__iter__"):
+            if not isinstance(results, Generator):
+                validated_plugin = self._validate_plugin(plugin_name, results)
+
+                if validated_plugin:
+                    yield validated_plugin
+
+            else:
                 # Only if it's an iterator, provider results as a series
```

Comment:
This comment is wrong?
Because the `if` statement above handles generators, which are iterators.

Should this be:

```suggestion
                # Only if it's an tuple, provider results as a series
```

IDK, I am a little confused here...

Reply:
Are we still handling tuples correctly??

Reply:
I believe so, I inverted the logic here, moving the Tuple handling up and pushing down the iterator logic.